analyzers/src/SonarAnalyzer.Common/Json/Parsing/LexicalAnalyzer.cs

```diff
@@ -129,25 +130,78 @@ private void NextPosition(bool checkEOI = true)
             }
             else
             {
-                do
-                {
-                    line++;
-                }
-                while (line < lines.Count && lines[line].Length == 0);
-                position = 0;
-                if (checkEOI && !CanReadCurrentChar)
-                {
-                    throw new JsonException("Unexpected EOI" + AtLocationSuffix);
-                }
+                SeekToNextLine(checkEOI);
+            }
+        }
+
+        private void NthPosition(int n, bool checkEOI = true)
+        {
+            for (var i = 0; i < n; i++)
+            {
+                NextPosition(checkEOI);
+            }
+        }
+
+        private void SeekToNextLine(bool checkEOI)
+        {
+            do
+            {
+                line++;
+            }
+            while (line < lines.Count && lines[line].Length == 0);
+            position = 0;
+            if (checkEOI && !CanReadCurrentChar)
+            {
+                throw new JsonException("Unexpected EOI" + AtLocationSuffix);
             }
         }
 
         private void SkipWhiteSpace()
         {
-            while (CanReadCurrentChar && char.IsWhiteSpace(CurrentChar))
+            var isInMultiLineComment = false;
+            while (CanReadCurrentChar)
             {
-                NextPosition(false);
+                if (IsWhitespace())
+                {
+                    NextPosition(false);
+                }
+                else if (IsSingleLineComment())
+                {
+                    // We just need to read starting from the next line
+                    SeekToNextLine(false);
+                }
+                else if (IsCloseMultiLineComment())
```

Comment:
`Close` resonatest as `Near` here a little
```suggestion
                else if (IsEndOfMultiLineComment())
```